include/dsn/c/app_model.h

```diff
@@ -292,51 +292,54 @@ enum dsn_chkpt_apply_mode
     DSN_CHKPT_LEARN
 };
 
-typedef dsn_error_t (*dsn_app_checkpoint)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef void(*dsn_app_on_batched_rpc_requests)(
+    void*,           ///< context from dsn_app_create
+    int64_t,         ///< ballot
+    int64_t,         ///< decree
+    dsn_message_t*,  ///< request array ptr
+    int              ///< request count
+    );
+
+typedef int(*dsn_app_get_internal_error)(
+    void*     ///< context from dsn_app_create
     );
 
-typedef dsn_error_t(*dsn_app_checkpoint_async)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef dsn_error_t(*dsn_app_sync_checkpoint)(
```

Comment:
but checkpoint related methods are a lot:
sync_checkpoint
async_checkpoint
get_last_checkpoint_decree
prepare_get_checkpoint
get_checkpoint
apply_checkpoint

I think rename them as "checkpoint_xxx" to just make them grouped will make
the code ugly.

On Mon, Jun 20, 2016 at 3:40 PM, Zhenyu Guo <notifications@github.com>
wrote:

> In include/dsn/c/app_model.h
> <https://github.com/imzhenyu/rDSN/pull/403#discussion_r67647016>:
>
> >      );
> >
> > -typedef dsn_error_t(*dsn_app_checkpoint_async)(
> > -    void*,  ///< context from dsn_app_create
> > -    int64_t ///< checkpoint version
> > +typedef dsn_error_t(*dsn_app_sync_checkpoint)(
>
> This does not contradict to the previous reasoning. You may use
> checkpoint_xxx as the original method so that the pattern is kept
> consistently and the methods are also grouped automatically during
> auto-completion.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/403/files/89a41d5e513a99014291d1ad76187ae04459b1ab#r67647016>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPVVebsZQ23C3Tkau_8cDw3UJmo-7ks5qNkPugaJpZM4I2J8D>
> .
>